packages/web-ui-registration/src/RegisterForm.tsx

```diff
@@ -157,14 +156,15 @@ export const RegisterForm = ({ setLoginRoute }: { setLoginRoute: DispatchLoginRo
 									placeholder={passwordConfirmationPlaceholder}
 								/>
 							</Field.Row>
-							{errors.passwordConfirmation?.type === 'validate' && (
-								<Field.Error>{t('registration.component.form.invalidConfirmPass')}</Field.Error>
-							)}
-							{errors.passwordConfirmation?.type === 'required' && (
-								<Field.Error>{t('registration.component.form.requiredField')}</Field.Error>
-							)}
-						</Field>
-					)}
+						)}
+						{errors.passwordConfirmation?.type === 'validate' && requiresPasswordConfirmation && (
+							<Field.Error>{t('registration.component.form.invalidConfirmPass')}</Field.Error>
+						)}
+						{errors.passwordConfirmation?.type === 'required' && requiresPasswordConfirmation && (
+							<Field.Error>{t('registration.component.form.requiredField')}</Field.Error>
+						)}
+						{passwordVerifications && <PasswordVerifier password={watch('password')} passwordVerifications={passwordVerifications} />}
+					</Field>
```

Comment:
I suggest adding the default value of the translation `onboarding.form.adminInfoForm.fields.password.placeholder` and `Confirm_new_password`, otherwise looks weird without it since the default value is empty.
![image](https://github.com/RocketChat/Rocket.Chat/assets/20212776/73c7b1d3-0cd0-41ab-96ed-b70db0afe99e)